python/cucim/src/cucim/core/operations/morphology/_pba_2d.py

```diff
@@ -1,8 +1,40 @@
+import functools
 import math
+import numbers
 import os
 
+
 import cupy
 
+try:
+    # math.lcm was introduced in Python 3.9
+    from math import lcm
+except ImportError:
+
+    """Fallback implementation of least common multiple (lcm)
+
+    TODO: remove once minimum Python requirement is >= 3.9
+    """
+
+    def _lcm(a, b):
+        return abs(b * (a // math.gcd(a, b)))
```

Comment:
NumPy also has an `lcm` implementation FWIW

Reply:
Okay, I didn't know that. This way is not quite as fast as `math.lcm` for scalar `a`, `b` but still faster than `numpy.lcm` in that case.